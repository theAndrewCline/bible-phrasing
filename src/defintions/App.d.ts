export type Passage = string
export type Passages = Passage[]