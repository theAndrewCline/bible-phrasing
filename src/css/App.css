@import url('https://fonts.googleapis.com/css?family=PT+Serif:400,400i,700');

h1 {
  color: #006064;
}

.App {
  text-align: center;
  font-family: 'PT Serif', 'serif';
}
