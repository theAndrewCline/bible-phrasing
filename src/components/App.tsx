import React, { Component } from 'react'
import NavBar from './NavBar'
import '../css/App.css'
import { Provider } from 'react-redux'
import { Store } from 'redux'
import { fetchPassage } from '../js/actions/actions';

// const store: Store = createStore()

class App extends Component<{}, {passages: any[]}> {
  constructor (props: any) {
    super(props)

    this.state = {
      passages: []
    }
  }

  public componentWillMount () {
    fetchPassage('John+3').then(passages => { 
      this.setState({passages})
    })
  }
  public render () {
    return (
      // <Provider store={store}>
      <div className='App'>
        <NavBar />
        <h1>John 1</h1>
        {this.state.passages.map((passage, i) => <p key={i}>{passage}</p>)}
      </div>
      // </Provider>
    )
  }
}

export default App
