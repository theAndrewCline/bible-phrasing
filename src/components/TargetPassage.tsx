import React from 'react'
import GridLayout from 'react-grid-layout'

import { Passages } from '../defintions/App'

type Props = {
  passages: Passages
}

function TargetPassage ({ passages }: Props) {

  return (
    <GridLayout className="layout" cols={1} rowHeight={8} width={1200} autoSize={true }>
      {passages.map((passage, i) => (

        <div key={i} data-grid={{x: 1, y: 1, w: 800, h: 1}}>{passage}</div>

      ))}
    </GridLayout>
  )

}

export default TargetPassage