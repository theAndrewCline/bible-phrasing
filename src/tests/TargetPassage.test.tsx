import React from 'react'
import { shallow } from 'enzyme'
import ToJson from 'enzyme-to-json'

import TargetPassage from '../components/TargetPassage'
import { Passages } from '../defintions/App';

describe('<TargetPassage/>', () => {
  const mockPassages: Passages = []
  it('should matchsnapshot', () => {
    const wrap = shallow(<TargetPassage passages={mockPassages} />)
    expect(ToJson(wrap)).toMatchSnapshot()
  })
})