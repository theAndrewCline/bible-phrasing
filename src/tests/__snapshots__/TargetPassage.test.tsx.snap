// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TargetPassage/> should matchsnapshot 1`] = `
<ReactGridLayout
  autoSize={true}
  className="layout"
  cols={1}
  compactType="vertical"
  containerPadding={null}
  draggableCancel=""
  draggableHandle=""
  isDraggable={true}
  isResizable={true}
  layout={Array []}
  margin={
    Array [
      10,
      10,
    ]
  }
  maxRows={Infinity}
  onDrag={[Function]}
  onDragStart={[Function]}
  onDragStop={[Function]}
  onLayoutChange={[Function]}
  onResize={[Function]}
  onResizeStart={[Function]}
  onResizeStop={[Function]}
  preventCollision={false}
  rowHeight={8}
  style={Object {}}
  useCSSTransforms={true}
  verticalCompact={true}
  width={1200}
/>
`;
